ruby_programming/test_driven_development/rspec_introduction_lesson.md

```diff
@@ -1,26 +1,245 @@
+
+The reason we choose RSpec is ***readability***. The simplicity of this choice, however, is a proxy for a deeper goal. Because RSpec is foolishly easy to read and write, it's easy to pick-up and the intentions and meanings of your tests are immediately clear (aka, ***transparent***). RSpec's readability and transparency, in turn, lend themselves to ***maintainability***. Code that is readable and transparent is easier to maintain, and a testing suite that is equally so ensures that your application as whole remains adaptable to inevitable change.
+
+But enough proselytizing. Strap your helmet and buckle up; we're going to jump right in.
+
+#### Installing RSpec
+
+Boot up your terminal and punch in `gem install rspec` to install RSpec. Once that's done, you can verify your version of RSpec with `rspec --version`, which will output the current version of each of the packaged gems. Take a minute also to hit `rspec --help` and look through the various options available. We'll come back to those later.
+
+Finally, `cd` into a project directory that you wish to configure for use with RSpec and type `rspec --init` to initialize RSpec within the project. This will generate `.rspec` and `spec/spec_helper.rb`, such that your project might look like:
+
+```
+project
+  |__lib
+  |   |__script.rb
+  |
+  |__spec
+  |   |__spec_helper.rb
+  |
+  |__.spec
+```
+
+That's it. Within two steps, you're up and running with RSpec. That was so hard, wasn't it?
+
+#### Basic syntax
+
+How 'bout a test to see the syntax? Create a new directory called "ruby_testing", change into it, and initiate RSpec.
+
+```
+$ mkdir ruby_testing
+$ cd ruby_testing
+$ rspec --init
+```
+
+As expected, the output will read:
+
+```bash
+  create   .rspec
+  create   spec/spec_helper.rb
+```
+
+Run the tests with `rspec`, which will return "No examples found." That really shouldn't surprise you, because we haven't written any tests yet. If you're still shocked...maybe take a short break, or come say hello to us in the Ruby Gitter room.
+
+```
+No examples found.
+
+
+Finished in 0.00037 seconds (files took 0.21108 seconds to load)
+0 examples, 0 failures
+```
+
+Let's add our first test. Let's say we want to create a calculator with a few methods (or behaviors) that we want to test. True to BDD, we will write the tests prior to the code. The `spec/` folder is where all your tests will live. Using `touch` on the command line or through your text editor, create `calculator_spec.rb` within the `spec/` folder and add the following lines:
```

Comment:
Are we covering BDD prior to this anywhere? This may confuse otherwise.

Reply:
Good point. I believe the previous lesson is an intro to TDD (which hopefully will also introduce BDD?)